### PR TITLE
Add custom shader_version on glow renderer

### PR DESCRIPTION
**Pull request by ar37-rs**
* Fix black screen by changing shader_version to OpenGL ES 2.0 shader (ES100) on VirtualBox VMSVGA driver with OpenGL ES 2.0 and OpenGL 2.1 with no srgb support.
* More easier cross compiling for different targets.

### CODE REVIEW

File: crates/eframe/CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ NOTE: [`egui-winit`](../egui-winit/CHANGELOG.md), [`egui_glium`](../egui_glium/C
 * Added `NativeOptions::event_loop_builder` hook for apps to change platform specific event loop options ([#1952](https://github.com/emilk/egui/pull/1952)).
 * Enabled deferred render state initialization to support Android ([#1952](https://github.com/emilk/egui/pull/1952)).
 * Allow empty textures with the glow renderer.
-
+* Added `custom_shader_version` to `NativeOptions` for cross compilling support on different target OpenGL | ES versions (on native `glow` renderer only).
```
Comment (emilk) [inline crates/eframe/CHANGELOG.md:11]:
```suggestion
* Added `custom_shader_version` to `NativeOptions` for cross compilling support on different target OpenGL | ES versions (on native `glow` renderer only) ([#1993](https://github.com/emilk/egui/pull/1993))..
```

File: crates/egui_glow/CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ All notable changes to the `egui_glow` integration will be noted in this file.
 
 ## Unreleased
 * Allow empty textures.
-
+* small BREAKING change: Added `custom_shader_version` variable on `EguiGlow::new` for easier cross compilling on different OpenGL | ES targets.
```
Comment (emilk) [inline crates/egui_glow/CHANGELOG.md:7]:
```suggestion
* Added `custom_shader_version` variable on `EguiGlow::new` for easier cross compilling on different OpenGL | ES targets ([#1993](https://github.com/emilk/egui/pull/1993)).
```

File: crates/egui_glow/src/winit.rs [hunk 3]
```diff
@@ -14,11 +14,13 @@ pub struct EguiGlow {
 }
 
 impl EguiGlow {
+    /// for automatic shader version detection set `custom_shader_version` to None
```
Comment (emilk) [inline crates/egui_glow/src/winit.rs:17]:
```suggestion
    /// For automatic shader version detection set `custom_shader_version` to `None`.
```

File: crates/egui_glow/src/painter.rs [hunk 4]
```diff
@@ -106,13 +106,16 @@ impl Painter {
...
         crate::profile_function!();
         crate::check_for_gl_error_even_in_release!(&gl, "before Painter::new");
 
         let max_texture_side = unsafe { gl.get_parameter_i32(glow::MAX_TEXTURE_SIZE) } as usize;
-
-        let shader_version = ShaderVersion::get(&gl);
+        let shader_version = match custom_shader_version {
+            Some(version) => version,
+            _ => ShaderVersion::get(&gl),
+        };
```
Comment (emilk) [inline crates/egui_glow/src/painter.rs:118]:
```suggestion
        let shader_version = custom_shader_version.uwrap_or_else(|| ShaderVersion::get(&gl));
```

File: crates/eframe/src/epi.rs [hunk 5]
```diff
@@ -311,6 +311,14 @@ pub struct NativeOptions {
     ///
     /// Note: A [`NativeOptions`] clone will not include any `event_loop_builder` hook.
     pub event_loop_builder: Option<EventLoopBuilderHook>,
+
+    #[cfg(feature = "glow")]
+    /// A helper when cross compiling for different target OpenGL | ES versions.
+    ///
+    /// cross compiling for VirtualBox VMSVGA driver with OpenGL 2.1 but doesn't support SRGB texture.
+    ///
+    /// the OpenGL ES 2.0 shader: ShaderVersion::Es100 can be used to solve blank texture problem (fallback shader).
```
Comment (emilk) [inline crates/eframe/src/epi.rs:320]:
```suggestion
    /// For OpenGL ES 2.0: set this to [`ShaderVersion::Es100`] to solve blank texture problem (by using the "fallback shader").
```

File: crates/eframe/src/epi.rs [hunk 6]
```diff
@@ -311,6 +311,14 @@ pub struct NativeOptions {
     ///
     /// Note: A [`NativeOptions`] clone will not include any `event_loop_builder` hook.
     pub event_loop_builder: Option<EventLoopBuilderHook>,
+
+    #[cfg(feature = "glow")]
+    /// A helper when cross compiling for different target OpenGL | ES versions.
+    ///
+    /// cross compiling for VirtualBox VMSVGA driver with OpenGL 2.1 but doesn't support SRGB texture.
```
Comment (emilk) [inline crates/eframe/src/epi.rs:318]:
```suggestion
    /// Needed fo cross compiling for VirtualBox VMSVGA driver with OpenGL 2.1 which doesn't support SRGB texture.
    /// See <https://github.com/emilk/egui/pull/1993>.
```

**Review by emilk — CHANGES_REQUESTED** [review]
Looks good, but I think `custom_shader_version` is better named as just `shader_version`.

**Comment by ar37-rs** [comment]
> Looks good, but I think `custom_shader_version` is better named as just `shader_version`.

@emilk thank you, done changed,

**Pull request merged**